### ISSUE TITLE
Pose-S3 output dimension

### ISSUE THREAD

**Issue by timchiu9781** [Xinyu-Yi/TransPose#25]
Hello Xinyu,
I have a question about the dimension of Pose-S3 output.
As mentioned in the paper, 
![image](https://user-images.githubusercontent.com/55866531/176634235-0a12528b-e99e-417c-9727-921faa8abe3b.png)
The output of Pose-S3 is R6(j-1), and j is 24. So the output dimension is 138
But the in the code, net.py, The pose_s3 output dimension is joint_set.n_reduced * 6, which is 90.
I have no idea about it.
Can you help me to figure out why the dimension is it?

So, when I want to calculate the loss of Pose-S3. I need to use the func:_reduced_glb_6d_to_full_local_mat. to get the full joint rotation matrix.
And use func:rotation_matrix_to_r6d, to get the full joint rotation in 6d, right?

**Comment by Xinyu-Yi** [comment]
For the loss, we directly use L2 on 6D representation, i.e., not converting to rotation matrices. It's a very simple implementation. You can try different losses.

**Comment by timchiu9781** [comment]
Thanks for reply Xinyu,

As you said, the output of Pose-S3 is actually 15d except L R ankle, foot, wrist, hand.
So when we calculate the loss in this layer, it just use the 15 point to calculate the loss with groundtruth.
Due to the DIP-IMU dataset did not have these information for training and evaluation.

But I have a concern about this,  the training dataset contain AMASS and partial DIP-IMU. And the AMASS have all joint information in the dataset. So you mean the Pose-S3 do not calculate the loss cause the DIP-IMU have no such information.

However, the AMASS have the information. And the network output is set to 90d. Does it mean that the network will not regress the other 8 joint rotation while training? And won't it lead the model have a poor performance reletivity?


Thanks for replying me so many questions.

**Comment by Xinyu-Yi** [comment]
1. These 8 joints' rotations are set to identity. As we put our imu on forearms and lower legs, by nature the wrist/hand/ankle/foot joint movements are not measured. Your method to calculate the ground truth is right.
2. In the name  "_reduced_glb_6d_to_full_local_mat", "global" means "root-relative", "local" means "parernt-relative"

**Comment by timchiu9781** [comment]
Hello Xinyu,
Thanks for your explanation. I understand what things are going.
